### DISCUSSION TITLE
模型是否有透明度属性可以设置

### DISCUSSION THREAD

**Discussion by KingCyril** [galacean/discussion#22] [Engine] [answered]
模型加载完毕后，转场时透明度设置由0到1慢慢显示如何实现

**Comment by singlecoder** [comment]
可以试试修改模型材质的 baseColor 的 alpha

**Reply by leaon4** [reply]
试了，不行。three.js是有转门的属性来控制的: transparent, opacity

**Comment by KingCyril** [comment, marked as answer]
我通过修改模型材质的 baseColor 的 alpha 但是渲染有问题，发现了官网中这段介绍：

1. 透明渲染异常？
请先确保材质开启了透明度模式，即材质的 [isTransparent](https://oasisengine.cn/0.8/api/core/BaseMaterial#isTransparent) 属性设置为了 true。
相应的材质的 [baseColor](https://oasisengine.cn/0.8/api/core/BlinnPhongMaterial#baseColor) 需要设置正确的透明度。如 material.baseColor.a = 0.5。透明度范围为 【0 ～ 1】，数值越小，越透明。
如果透明度渲染仍有异常，请确保材质的颜色混合度模式（[blendMode](https://oasisengine.cn/0.8/api/core/BaseMaterial#blendMode)）为期望的组合。
有一些透明度渲染异常可能是因为没有关闭背面剔除，可以通过 [renderFace](https://oasisengine.cn/0.8/api/core/BaseMaterial#renderFace) 来设置想要渲染的面。

我想请问一下《颜色混合度模式为期望的组合》和《设置想要渲染的面》这几个概念怎么去理解和设置调试去使模型渲染正常

**Reply by singlecoder** [reply]
### 颜色混合度模式为期望的组合
颜色混合模式建议看看 https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/blendFuncSeparate

对应到引擎的实现可以看看 RenderState，参考代码如下：
```typescript
private _createSpriteMaterial(): Material {
  const material = new Material(this, Shader.find("Sprite"));
  const renderState = material.renderState;
  const target = renderState.blendState.targetBlendState;
  target.enabled = true;
  target.sourceColorBlendFactor = BlendFactor.SourceAlpha;
  target.destinationColorBlendFactor = BlendFactor.OneMinusSourceAlpha;
  target.sourceAlphaBlendFactor = BlendFactor.One;
  target.destinationAlphaBlendFactor = BlendFactor.OneMinusSourceAlpha;
  target.colorBlendOperation = target.alphaBlendOperation = BlendOperation.Add;
  renderState.depthState.writeEnabled = false;
  renderState.rasterState.cullMode = CullMode.Off;
  material.renderQueueType = RenderQueueType.Transparent;
  material.isGCIgnored = true;
  return material;
}
```

### 设置想要渲染的面
关于渲染面，建议看看 https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/cullFace

对应的引擎使用方式如下：
```typescript
renderState.rasterState.cullMode = CullMode.Off;
```

**Reply by KingCyril** [reply]
有没有demo啊，光看属性有点懵

**Reply by zhuxudong** [reply]
* 设置 isTransparent
* 设置渲染面, renderState.rasterState.cullMode = CullMode.Off;
* 设置透明度, baseColor.a =0.5;

**Reply by singlecoder** [reply]
补充下透明度设置：

material. isTransparent

**Comment by KingCyril** [comment]
https://img.alicdn.com/imgextra/i3/O1CN01TvcpI41qb0I4R6Yyj_!!6000000005513-2-tps-268-870.png

https://img.alicdn.com/imgextra/i4/O1CN014GOIVq1Lam069irJ1_!!6000000001316-2-tps-279-848.png

麻烦帮忙看一下这个模型透明度为啥渲染成这样子了

**Reply by singlecoder** [reply]
来一个你的最小示例呗～
